src/classes/astandardpaths.cpp

```diff
@@ -1,51 +1,37 @@
 #include "src/classes/astandardpaths.h"
 
-QMap<AStandardPaths::Directories, QString> AStandardPaths::directories;
+QMap<AStandardPaths::Directories, QDir> AStandardPaths::directories;
 
 void AStandardPaths::setup()
 {
     auto data_location = QStandardPaths::AppDataLocation;
     directories = { // [F]: Dir could be ambiguous since it is very similar to QDir
-        {Database, QDir::toNativeSeparators(
-         QStandardPaths::writableLocation(data_location) + '/')},
-        {Templates, QDir::toNativeSeparators(
-         QDir(QStandardPaths::writableLocation(data_location)).filePath(QStringLiteral("templates/")))},
-        {DatabaseBackup, QDir::toNativeSeparators(
-         QDir(QStandardPaths::writableLocation(data_location)).filePath(QStringLiteral("backup/")))}
+        {Database, QDir(QStandardPaths::writableLocation(data_location) + '/')},
+        {Templates, QDir(QStandardPaths::writableLocation(data_location)).filePath(
+         QStringLiteral("templates/"))},
+        {Backup, QDir(QStandardPaths::writableLocation(data_location)).filePath(
+         QStringLiteral("backup/"))}
```

Comment:
Quick: `/` are redundant.

This code is suspect. `QDir("...").filePath("...")` returns a QString however a QDir with *that* string is created. The QString is somehow implicitely used for the QDir construction. I still believe my original idea is still the simplest explicit one
```c++
{Database, QDir(QStandardPaths::writableLocation(data_location))},  // no / needed
{Templates, QDir(QStandardPaths::writableLocation(data_location) + QStringLiteral("/templates"))},  // here it is at the start
{Backup, QDir(QStandardPaths::writableLocation(data_location) + QStringLiteral("/backup"))}
```
The QDir is explicitelly created through its consructor with no funny business

Reply:
I'm gonna be honest with you, that went over my head.

But I see the benefits of your approach, commit coming up!